### PR TITLE
MM-11603 Add a retry screen when fetching teams fails

### PR DESCRIPTION
**Pull request by sudheerDev**
#### Summary
Add an error screen when for failure in fetching teams

#### Ticket Link
[MM-11603](https://mattermost.atlassian.net/browse/MM-11603)
#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes

#### Device Information
This PR was tested on: Emulators

#### Screenshots
sep-07-2018 02-59-25
sep-07-2018 02-58-51

### CODE REVIEW

File: app/screens/error_teams_list/error_teams_list.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                statusBarHideWithNavBar: false,
+                screenBackgroundColor: theme.centerChannelBg,
+            },
+        });
+    };
+
+    getUserInfo = async () => {
+        this.setState({loading: true});
+        this.props.actions.connection(true);
+        await this.props.actions.loadMe();
```
Comment (sudheerDev) [inline app/screens/error_teams_list/error_teams_list.js]: `loadMe` always returns {data: true} so there is no point in checking for an error and if there is an error `selectDefaultTeam` from root will again send back to this screen as shown in the first gif for retry

File: app/screens/error_teams_list/error_teams_list.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                statusBarHidden: false,
+                statusBarHideWithNavBar: false,
+                screenBackgroundColor: theme.centerChannelBg,
+            },
+        });
+    };
+
+    getUserInfo = async () => {
+        this.setState({loading: true});
+        this.props.actions.connection(true);
```
Comment (sudheerDev) [inline app/screens/error_teams_list/error_teams_list.js]: There are chances that this is set false so network requests wont go through so manually setting this to `true`

**Comment by sudheerDev** [comment]
@enahum First screenshot i killed my local server and retrying with that.

second screenshot before retrying I started my server again

**Comment by enahum** [comment]
Cool, can you create a ticket to improve the UX when the channel fails to load? That seems to be the second screenshot that does not say much really ;)

Thanks

**Comment by sudheerDev** [comment]
Sure but should we do this for other places as well? Like mentions etc?

**Comment by enahum** [comment]
Recent mentions and flagged post already use the same UI as you are using in the teams screen

**Comment by sudheerDev** [comment]
Aah. ya that could be in the same component but this needs slight changes to make it work in the same way as there is no error callback for `loadMe` call. Anyway will add these details as scope of the ticket.

**Pull request merged**